operators/src/source/gdal_source/mod.rs

```diff
@@ -144,6 +161,95 @@ pub struct GdalDatasetGeoTransform {
     pub y_pixel_size: f64,
 }
 
+impl GdalDatasetGeoTransform {
+    pub fn standardized_spatial_partition(
+        &self,
+        x_size: usize,
+        y_size: usize,
+    ) -> SpatialPartition2D {
+        // The geo transform of a gdal dataset might have flipped axis. So correct that.
+
+        // the opposite y value (y value of the non origin edge)
+        let opposite_coord_y = self.origin_coordinate.y + self.y_pixel_size * y_size as f64;
+
+        // if the y-axis is negative then the origin is on the upper side.
+        let (upper_y, lower_y) = if self.y_pixel_size.is_sign_negative() {
+            (self.origin_coordinate.y, opposite_coord_y)
+        } else {
+            (opposite_coord_y, self.origin_coordinate.y)
+        };
+
+        let opposite_coord_x = self.origin_coordinate.x + self.x_pixel_size * x_size as f64;
+
+        // if the y-axis is negative then the origin is on the upper side.
+        let (left_x, right_x) = if self.x_pixel_size.is_sign_positive() {
+            (self.origin_coordinate.x, opposite_coord_x)
+        } else {
+            (opposite_coord_x, self.origin_coordinate.x)
+        };
+
+        SpatialPartition2D::new_unchecked(
+            Coordinate2D::new(left_x, upper_y),
+            Coordinate2D::new(right_x, lower_y),
+        )
+    }
+
+    /// Transform a `Coordinate2D` into a `GridIdx2D`
+    #[inline]
+    pub fn coordinate_to_grid_idx_2d(&self, coord: Coordinate2D) -> GridIdx2D {
+        // TODO: use an epsilon here?
+        let grid_x_index =
+            ((coord.x - self.origin_coordinate.x) / self.x_pixel_size).floor() as isize;
+        let grid_y_index =
+            ((coord.y - self.origin_coordinate.y) / self.y_pixel_size).floor() as isize;
+
+        [grid_y_index, grid_x_index].into()
+    }
+
+    fn spatial_partition_to_read_window(
+        &self,
+        spatial_partition: &SpatialPartition2D,
+    ) -> GdalReadWindow {
+        // choose the epsilon relative to the pixel size
```

Comment:
maybe explain why you do this. I guess in order to not read the next pixel, because gdals reads inclusive and our partitions are exlusive? If so we should need the epsilon only on the bottom right corner.

Reply:
[8fd3b20](https://github.com/geo-engine/geoengine/pull/602/commits/8fd3b20b83382d3b413047c666ec86f744e593b1)

Reply:
[313204f](https://github.com/geo-engine/geoengine/pull/602/commits/313204fa2ba923767410b29e2bbc6bafc5014a92)